util/rate_limiter_test.cc

```diff
@@ -117,12 +119,26 @@ TEST_F(RateLimiterTest, Rate) {
               arg.request_size - 1, target / 1024, rate / 1024,
               elapsed / 1000000.0);
 
-      ASSERT_GE(rate / target, 0.80);
+      ++samples;
+      if (rate / target >= 0.80) {
+        ++samples_at_minimum;
+      }
       ASSERT_LE(rate / target, 1.25);
     }
   }
-}
+
+  // This can fail in heavily loaded CI environments
+  if (getenv("SANDCASTLE")
+#if (defined(TRAVIS) || defined(CIRCLECI)) && defined(OS_MACOSX)
+      || true
 #endif
```

Comment:
This might be clearer as two separate conditions rather than the || true

Reply:
IMHO `#else` adds extra complication.

Reply:
How about something like

```
#if (defined(TRAVIS) || defined(CIRCLECI)) && defined(OS_MACOSX)
#define DISABLE_MINIMUM_RATE_CHECK true
#else
#define DISABLE_MINIMUM_RATE_CHECK false
#endif
```

and then

```
if (DISABLE_MINIMUM_RATE_CHECK || getenv("SANDCASTLE")) { ... }
```

?